src/components/PostDetails.tsx

```diff
@@ -5,27 +5,48 @@ import { NewCommentForm } from './NewCommentForm';
 import * as commentsApi from '../api/comments';
 
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+// eslint-disable-next-line max-len
+import {
+  addComments,
+  deleteCommentById,
+  allComments,
+  isLoaded,
+  hasCommentsError,
+  setIsLoaded,
+  setHasError,
+} from '../features/comments/commentsSlice';
 
 type Props = {
   post: Post;
 };
 
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const comments = useAppSelector(allComments);
+  const loaded = useAppSelector(isLoaded);
+
+  const hasError = useAppSelector(hasCommentsError);
+
+  // const [comments, setComments] = useState<Comment[]>([]);
+  // const [loaded, setLoaded] = useState(false);
+  // const [hasError, setError] = useState(false);
   const [visible, setVisible] = useState(false);
 
   function loadComments() {
-    setLoaded(false);
-    setError(false);
+    // setLoaded(false);
+    // setError(false);
+    dispatch(setIsLoaded(false));
+    dispatch(setHasError(false));
     setVisible(false);
 
     commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
+      .then(res => {
+        dispatch(addComments(res));
+      }) // save the loaded comments
+      .catch(() => dispatch(setHasError(true))) // show an error when something went wrong
+      .finally(() => dispatch(setIsLoaded(true))); // hide the spinner
```

Comment:
It's not the way you handle queries in Redux. As I told, read more about `createAsyncThunk`.